DMCompiler/DMCompiler.cs

```diff
@@ -211,6 +212,13 @@ public bool Emit(WarningCode code, Location loc, string message) {
                 break;
         }
 
+        if (Settings.StoreMessages)
+            if (CompilerMessages.Count < 1000)
+                CompilerMessages.Add(new CompilerEmission(ErrorLevel.Error, loc, message).ToString());
```

Comment:
Is there a reason this is creating a new CompilerEmission that's always an error instead of using the already created `emission`?